ui/src/message_popup/scaffolding_card.jsx

```diff
@@ -1,61 +1,149 @@
 import React from "react";
 import Toggle from 'material-ui/Toggle';
 import Divider from 'material-ui/Divider';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+import TextChangeEvent from '../types/dom_types.js';
 import {RadioButton, RadioButtonGroup} from 'material-ui/RadioButton';
 import Slider from 'material-ui/Slider';
 import _ from 'lodash';
 
+import {learningObjectives} from '../data/learning_objectives.js';
+import {allQuestions} from './questions.js';
+import {allStudents} from '../data/virtual_school.js';
 
 const ALL_COMPETENCY_GROUPS = 'ALL_COMPETENCY_GROUPS';
 
 export default React.createClass({
   displayName: "ScaffoldingCard",
   
   propTypes: {
-    competencyGroupValue: React.PropTypes.string.isRequired,
-    competencyGroups: React.PropTypes.array.isRequired,
-    onCompetencyGroupChanged: React.PropTypes.func.isRequired,
-    sessionLength: React.PropTypes.number.isRequired,
-    questions: React.PropTypes.array.isRequired,
-    onSliderChange: React.PropTypes.func.isRequired,
-    shouldShowStudentCard: React.PropTypes.bool.isRequired,
-    onStudentCardsToggled: React.PropTypes.func.isRequired,
-    shouldShowSummary: React.PropTypes.bool.isRequired,
-    onSummaryToggled: React.PropTypes.func.isRequired,
-    helpType: React.PropTypes.string.isRequired,
-    onHelpToggled: React.PropTypes.func.isRequired,
-    itemsToShow: React.PropTypes.object.isRequired
+    scaffolding: React.PropTypes.shape({
+      helpType: React.PropTypes.string.isRequired,
+      shouldShowStudentCard: React.PropTypes.bool.isRequired,
+      shouldShowSummary: React.PropTypes.bool.isRequired
+    }).isRequired,
+    email: React.PropTypes.string.isRequired,
+    itemsToShow: React.PropTypes.object.isRequired,
+    saveScaffoldingAndSession: React.PropTypes.func.isRequired
   },
   
+  getInitialState(){
+    const isSolutionMode = _.has(this.props.itemsToShow, 'solution');
+    return ({
+      email: this.props.email,
+      questions: allQuestions,
```

Comment:
We can remove `questions` from state altogether, since `getQuestions` uses the information in `competencyGroupValue` and `isSolutionMode` to compute them.